pkg/controllers/nodeclaim/disruption/controller.go

```diff
@@ -58,10 +61,11 @@ type Controller struct {
 // NewController constructs a nodeclaim disruption controller
 func NewController(clk clock.Clock, kubeClient client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider) operatorcontroller.Controller {
 	return operatorcontroller.Typed[*v1beta1.NodeClaim](kubeClient, &Controller{
-		kubeClient: kubeClient,
-		drift:      &Drift{cloudProvider: cloudProvider},
-		expiration: &Expiration{kubeClient: kubeClient, clock: clk},
-		emptiness:  &Emptiness{kubeClient: kubeClient, cluster: cluster, clock: clk},
+		kubeClient:           kubeClient,
+		supportedNodeClasses: cloudProvider.GetSupportedNodeClasses(),
```

Comment:
WDYT about just passing in the cloud provider as a variable? The way we do it in other controllers is to pass in the cloud provider so that we can just call whatever member functions we need. It's also how we create the Controller anyway.

Reply:
Make sense, just added that